runtime/interpreter/encode.go

```diff
@@ -755,13 +757,13 @@ func (e *Encoder) prepareDictionaryValue(
 			if err != nil {
 				return nil, err
 			}
-			entries[key] = prepared
+			entries = append(entries, prepared)
```

Comment:
Directly setting the value to the index might be slightly better in terms of performance (I haven't tested though)
```suggestion
			entries[index] = prepared
			index++
```

Reply:
VERY nice catch!  Your hunch is correct!  :+1:

Reply:
In isolation (small example program), this change provides a bigger improvement.  

For encoding, the new way required more coding changes.  Encoding speed improvement is around 0.5-0.7% vs ```append``` for large data.  Thanks again for the feedback, I'll incorporate this change!